### ISSUE TITLE
UniversalTypeClient6.munki can't make override

### ISSUE THREAD

**Issue by jannheider** [autopkg/jessepeterson-recipes#46]
Can't make autopkg override.
Fresh install of OSX 10.12 with munki and autopkg.

`autopkg make-override UniversalTypeClient6.munki`

Result:

> Traceback (most recent call last):
>   File "/usr/local/bin/autopkg", line 2435, in <module>
>     sys.exit(main(sys.argv))
>   File "/usr/local/bin/autopkg", line 2429, in main
>     exit(subcommands[verb]['function'](argv))
>   File "/usr/local/bin/autopkg", line 1787, in make_override
>     FoundationPlist.writePlist(override_plist, override_file)
>   File "/Library/AutoPkg/FoundationPlist/FoundationPlist.py", line 131, in writePlist
>     plistData = _plistToData(plistObject)
>   File "/Library/AutoPkg/FoundationPlist/FoundationPlist.py", line 106, in _plistToData
>     raise NSPropertyListSerializationException(error)
> FoundationPlist.FoundationPlist.NSPropertyListSerializationException: Error Domain=NSCocoaErrorDomain Code=3851 "Property list invalid for format: 100 (property lists cannot contain objects of type 'CFNull')" UserInfo={NSDebugDescription=Property list invalid for format: 100 (property lists cannot contain objects of type 'CFNull')}

**Comment by jessepeterson** [comment]
I'm not able to duplicate this error. Are you still getting it?

**Comment by jannheider** [comment]
Yes still having this issue
autopkg version : 1.0.2 OSX 10.12

**Comment by jessepeterson** [comment]
That's odd. Are you sure the recipe hasn't been modified and is up to take? Can you try on a "fresh" autopkg setup? My setup seems to do fine:

```
$ autopkg make-override UniversalTypeClient6.munki
Override file saved to /Users/jesse/Library/AutoPkg/RecipeOverrides/UniversalTypeClient6.munki.recipe
$
```

**Comment by jannheider** [comment]
Just installed autopkg on a new System and only added jessepeterson-recipes

`autopkg repo-add jessepeterson-recipes`

Same problem:

```
MacBookPro:~ testuser$ autopkg make-override UniversalTypeClient6.munki
Traceback (most recent call last):
  File "/usr/local/bin/autopkg", line 2435, in <module>
    sys.exit(main(sys.argv))
  File "/usr/local/bin/autopkg", line 2429, in main
    exit(subcommands[verb]['function'](argv))
  File "/usr/local/bin/autopkg", line 1787, in make_override
    FoundationPlist.writePlist(override_plist, override_file)
  File "/Library/AutoPkg/FoundationPlist/FoundationPlist.py", line 131, in writePlist
    plistData = _plistToData(plistObject)
  File "/Library/AutoPkg/FoundationPlist/FoundationPlist.py", line 106, in _plistToData
    raise NSPropertyListSerializationException(error)
FoundationPlist.FoundationPlist.NSPropertyListSerializationException: Error Domain=NSCocoaErrorDomain Code=3851 "Property list invalid for format: 100 (property lists cannot contain objects of type 'CFNull')" UserInfo={NSDebugDescription=Property list invalid for format: 100 (property lists cannot contain objects of type 'CFNull')}
```

Autopkg Version: 1.0.2
MacOS Version: 10.12.5

Override for UniversalTypeClient5.munki is working fine.

**Comment by jannheider** [comment]
ok, i looked more into it.

You are Using in UniversalTypeClient6.munki
```
<key>Processor</key>
<string>com.github.scriptingosx.SharedProcessors/FileTemplate</string>
```
this processor is coming from "scriptingosx-recipes"

So just add "scriptingosx-recipes" and the override is working.

Maybe you could add a Documentation where you list the dependencies to other repros ?

**Comment by jessepeterson** [comment]
So if you try to make an override that happens to include a processor that you don't have the recipes for then autopkg throws an error? Honestly that sounds like a bug in autopkg — that's pretty terrible UI. :) But hypothetically where would you (or any user) look to find such documentation?

**Comment by jannheider** [comment]
> So if you try to make an override that happens to include a processor that you don't have the recipes for then autopkg throws an error? 

Exactly 

> Honestly that sounds like a bug in autopkg — that's pretty terrible UI. :)

It looks like this is actually normal behavior -> https://github.com/autopkg/autopkg/issues/83

> But hypothetically where would you (or any user) look to find such documentation?

Something like an README.md, i just made a pull request with an readme ;)

**Comment by jannheider** [comment]
Issue Closed for #48
